### ISSUE TITLE
function make_parameter_data_frames() in Make_parameter_sets.R

### ISSUE THREAD

**Issue by zhengjiji456** [JohannesOberpriller/SensitivityAnalysisLPJ#2]
Hi, thanks for updating the R codes. 
I met a bug when running the codes in lines 186-189. The datasets 'climaticranges' used here was loaded from "./EnvironmentalData/climateranges_list.rds". 
> climatic_ranges_site = matrix(ncol = length(climaticranges), nrow = 2)
> climatic_ranges_site
     [,1] [,2] [,3] [,4] [,5] [,6]
[1,]   NA   NA   NA   NA   NA   NA
[2,]   NA   NA   NA   NA   NA   NA
> climatic_ranges_site[,1] = climaticranges[[1]] # co2 for example
Error in climatic_ranges_site[, 1] = climaticranges[[1]] # the numbers of col and row did not match.

In addition, I did not find the codes for generating the datasets (e/g. climateranges_list.rds) in file EnvironmentalData.

**Comment by JohannesOberpriller** [comment]
Hi, yes thank you. I pushed a bug fix and want to apologize that I forgot to change this, as I changed the climateranges_list for another script. The bug fix worked for me after pulling into a fresh repo, but can you confirm?

Climate meta data files are infered from the climatic data. As we can not upload such big climate data to github, we provide the extracted values here. In principle you can choose whatever ranges you are interested in and which are plausible. 

As I was doing this analysis and the new improvements of the rlpjguess package hand in hand, I probably forgot about updating some scripts. Thank you again.

**Comment by zhengjiji456** [comment]
Hi, thanks for your quick reply.
Unfortunately, I tried but failed again. I am not sure I understand the structure of the data (i.e. climatic_ranges_site) correctly.
  
##lines 89-99
  site_id = mastersheet["site_id"]  ## site_id with 600 rows
  print(site_id)
  ## need this as an input later
  names_climatic = names(climaticranges)
  climatic_ranges_site = matrix(ncol = length(climaticranges), nrow = 2) ## only 6 cols and 2 rows
  rownames(climatic_ranges_site) = c("minimum", "maximum")
  for(i in 1:length(names_climatic)){
    climatic_ranges_site[,i] = climaticranges[[i]][2:3,site_id] ## each climaticranges[[i]] is 3*200, so site_id here would be out of bound
  }

**Comment by zhengjiji456** [comment]
Hi, thanks for your quick reply.
Unfortunately, I tried but failed again. I am not sure I understand the structure of the data (i.e. climatic_ranges_site) correctly.
  
##lines 89-99
  site_id = mastersheet["site_id"]  ## site_id with 600 rows？
  print(site_id)
## need this as an input later
  names_climatic = names(climaticranges)
  climatic_ranges_site = matrix(ncol = length(climaticranges), nrow = 2) ## only 6 cols and 2 rows？
  rownames(climatic_ranges_site) = c("minimum", "maximum")
  for(i in 1:length(names_climatic)){
    climatic_ranges_site[,i] = climaticranges[[i]][2:3,site_id] ## each climaticranges[[i]] is 3*200, so site_id here would be out of bound？
  }

**Comment by zhengjiji456** [comment]
Hi, thanks for your quick reply.
Unfortunately, I tried but failed again. I am not sure I understand the structure of the data (i.e. climatic_ranges_site) correctly.
  
##lines 89-99
  site_id = mastersheet["site_id"]  ## site_id with 600 rows
  print(site_id)
  
  names_climatic = names(climaticranges)
  climatic_ranges_site = matrix(ncol = length(climaticranges), nrow = 2) ## only 6 cols and 2 rows
  rownames(climatic_ranges_site) = c("minimum", "maximum")
  for(i in 1:length(names_climatic)){
    climatic_ranges_site[,i] = climaticranges[[i]][2:3,site_id] ## each climaticranges[[i]] is 3*200, so site_id here would be out of bound
  }

**Comment by JohannesOberpriller** [comment]
no, site_id runs 3 times from 1 to 200 as defined in the following lines. 


mastersheet[,c("site_id","lon","lat")] = sites[,c("Site","Longitudinal","Latitudinal")]

mastersheet[1:200,4:6] = matrix(rep(c(1,0,0),200),ncol = 3, nrow = 200, byrow = T )
mastersheet[201:400,4:6] = matrix(rep(c(0,1,0),200),ncol = 3, nrow = 200, byrow = T )
mastersheet[401:600,4:6] = matrix(rep(c(0,0,1),200),ncol = 3, nrow = 200, byrow = T )

so I pulled it into a clean repo and it worked again. Can you send me a reproducible example of code that fails?

**Comment by JohannesOberpriller** [comment]
Hi, the site_id runs 3 times from 1 to 200 see also the following lines of code:

## determine a mastersheet to get an order for runnning
mastersheet = data.frame(matrix(nrow = 600, ncol = 6))

colnames(mastersheet) = c("site_id","lon","lat","Fag_syl","Pic_abi", "Pin_syl")

mastersheet[,c("site_id","lon","lat")] = sites[,c("Site","Longitudinal","Latitudinal")]

mastersheet[1:200,4:6] = matrix(rep(c(1,0,0),200),ncol = 3, nrow = 200, byrow = T )
mastersheet[201:400,4:6] = matrix(rep(c(0,1,0),200),ncol = 3, nrow = 200, byrow = T )
mastersheet[401:600,4:6] = matrix(rep(c(0,0,1),200),ncol = 3, nrow = 200, byrow = T )

can you send me a reproducible example that fails?

**Comment by zhengjiji456** [comment]
Hi, please see the warning message below and check the attachment. 

> apply(X = mastersheet, FUN = make_parameter_data_frames, climaticranges = climaticranges, white_parameters = white_parameters_mono, parameter_range_list = parameter_list, MARGIN = 1)
site_id 
      1 
Error in
 gzfile(file, mode) : cannot open the connection
In addition: Warning message:
In gzfile(file, mode) :
 Rerun with Debug
 Error in gzfile(file, mode) : cannot open the connection

**Comment by zhengjiji456** [comment]
[Make_parameter_sets.RData.zip](https://github.com/JohannesOberpriller/SensitivityAnalysisLPJ/files/7901415/Make_parameter_sets.RData.zip)

**Comment by zhengjiji456** [comment]
when I check the loop in function make_parameter_data_frames() step by step, I found climatic_ranges_site[,i] = climaticranges[[i]][2:3,site_id] (line 98) with a error message invalid subscript type 'list'. 

Perhaps you may show me the structure of the variable climatic_ranges_site(how many rows and cols) used in the later simulation, and I would find the reason I met these errors.

**Comment by JohannesOberpriller** [comment]
Hi, "cannot open the connection" ususally means that you cannot write the file. I guess this is because i did not automatically create the repositories. 

I now added these lines of code and pushed them: 

dirs = sapply(FUN = function(x){file.path(getwd(),x)}, X = c("parameters",
                                                             "R_scripts",
                                                             "submit_scripts",
                                                             "grids"))
for(dir in dirs){
  if(!dir.exists(dir)){
  dir.create(dir)
  }
}

I looked at the rdata file, but this seems to be ok for me.
